File: ling c/Funcionario.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct {
    char nm[50];
    int id;
    char cg[50];
    double sl;
    struct Departamento* depto;
} Funcionario;

typedef struct Departamento {
    char nm[50];
    int id;
    char loc[50];
    Funcionario** funcs;
    int numFuncionarios;
} Departamento;

Funcionario* criarFuncionario(char nome[], int id, char cargo[], double salario, Departamento* departamento);
void atualizarInfoPessoais(Funcionario* funcionario, char nome[], char cargo[], double salario);
void transferirDepto(Funcionario* funcionario, Departamento* novoDepartamento);
Departamento* obterDeptoAssociado(Funcionario* funcionario);

Departamento* criarDepartamento(char nome[], int id, char localizacao[]);
void adicionarFuncionario(Departamento* departamento, Funcionario* funcionario);
void removerFuncionario(Departamento* departamento, Funcionario* funcionario);
Funcionario** obterFuncionarios(Departamento* departamento);

File: hs/Funcionario.hs
data Funcionario = Funcionario { nm :: String, id :: Int, cg :: String, sl :: Double, depto :: Departamento }

data Departamento = Departamento { nmD :: String, idD :: Int, loc :: String, funcs :: [Funcionario] }

criarFuncionario :: String -> Int -> String -> Double -> Departamento -> Funcionario
atualizarInfoPessoais :: Funcionario -> String -> String -> Double -> Funcionario
transferirDepto :: Funcionario -> Departamento -> Funcionario
obterDeptoAssociado :: Funcionario -> Departamento

criarDepartamento :: String -> Int -> String -> Departamento
adicionarFuncionario :: Departamento -> Funcionario -> Departamento
removerFuncionario :: Departamento -> Funcionario -> Departamento
obterFuncionarios :: Departamento -> [Funcionario]
